packages/build/src/error/types.ts

```diff
@@ -1,3 +1,77 @@
+export type BuildError = BasicErrorInfo & {
+  title: string
+  pluginInfo?: string
+  locationInfo?: string
+}
+
+export type BasicErrorInfo = {
+  message: string
+  stack: string
+  severity: string
+  type: ErrorTypes
+  errorInfo: ErrorInfo
+  errorProps: any
```

Comment:
Do we know anything at all about what this could be? Is it an object with arbitrary properties? If so, we could make this a bit more specific with something like `Record<string, unknown>`.

Reply:
Good point. I dind't know we could restrict further through records but `Record<string, unknown>` might be our best bet since this is coming from:
- https://github.com/netlify/build/blob/fd75258a36e9e47aeed8d1099f6c9f87d2957328/packages/build/src/error/parse/parse.js#L62
- https://github.com/netlify/build/blob/fd75258a36e9e47aeed8d1099f6c9f87d2957328/packages/build/src/error/info.js#L19-L26

And this is basically an arbitrary object at the end of the day, using the custom key we define and where we bundle in all the error info through:
- https://github.com/netlify/build/blob/fd75258a36e9e47aeed8d1099f6c9f87d2957328/packages/build/src/error/info.js#L10-L17

Reply:
Potentially later on if we type `addErrorInfo` and `getErrorInfo` (which I think it will be some version of the `BasicErrorInfo` I have over here ☝️), these types will also improve.

Reply:
Done via - https://github.com/netlify/build/pull/5374/commits/709815e87967faedcf2befc28b3ef368231bf137